### PR TITLE
New top nav, language selector and Russian translations

### PR DESCRIPTION
**Pull request by johnnyd-eth**
## What does this PR do and why?

- Design and implement a new top Nav which includes a language selector (also language links in footer) see https://github.com/jbx-protocol/juice-interface/pull/317 for diff and more details.
- Implement a language selector. See https://github.com/jbx-protocol/juice-interface/pull/309 for initial implementation which has since had its design updated in https://github.com/jbx-protocol/juice-interface/pull/317
- Implements all currently completed Russian translations and marks ~200 pieces of text in the code for translation. Of the translations that have been marked in the code, Crowdin tells us that 87% of the Russian translations have been translated. The remaining 13% is mostly pieces of text added or updated since the translations were made. We should hopefully have the remainder completed very shortly once Crowdin is pulled. (NOTE: No Crowdin work is included in this PR, it's in a separate PR https://github.com/jbx-protocol/juice-interface/pull/329)

## Screenshots or screen recordings

Desktop:

https://user-images.githubusercontent.com/96150256/148473157-fc6363eb-cead-4916-8ab4-0eeb8bb9a57b.mp4

Mobile:

https://user-images.githubusercontent.com/96150256/148473359-f263e1a9-e4fb-4744-bf5d-92260de65f6c.mp4

## Acceptance checklist

- [x] I have evaluted the [Approval Guidelines](../../CONTRIBUTING.md#approval-guidelines) for this PR.

### CODE REVIEW

**Review by peripheralist — APPROVED** [review]
exciting!

**Comment by peripheralist** [comment]
looking great @aeolianeth! only one thing—looks like a little formatting error got into the mods list: 
image

once we get that and the merge conflicts fixed let's ship!

**Comment by johnnyd-eth** [comment]
> looking great @aeolianeth! only one thing—looks like a little formatting error got into the mods list: image
> 
> once we get that and the merge conflicts fixed let's ship!

Thanks for that pickup @peripheralist! I've fixed that and resolved the merge conflicts.

**Comment by aeolianeth** [comment]
> looking great @aeolianeth!

looking great ~~@aeolianeth!~~ @johnnyd-eth 😉

File: src/components/Create/BudgetForm.tsx
Comment (aeolianeth) [inline src/components/Create/BudgetForm.tsx:70]: **issue**: we should avoid splitting sentences when adding translations. Luckily, linguiJS allows us to wrap this whole block in a `<Trans>`:

```suggestion
<Trans>
          <a
          href={helpPagePath('protocol/learn/topics/funding-cycle')}
          rel="noopener noreferrer"
          target="_blank"
        >
          Learn more
         </a>
         about funding cycles.
         </Trans>
```

File: src/components/Create/BudgetForm.tsx
Comment (aeolianeth) [inline src/components/Create/BudgetForm.tsx:98]: **issue**: lets wrap this whole block in `Trans`

File: src/components/Create/BudgetForm.tsx
Comment (aeolianeth) [inline src/components/Create/BudgetForm.tsx:123]: **praise**: for not including the extra `{' '}` in the translation!

File: src/components/Create/BudgetForm.tsx
Comment (aeolianeth) [inline src/components/Create/BudgetForm.tsx:159]: **issue**: lets wrap the whole block in `trans` to avoid splitting sentence.

File: src/components/Create/RulesForm.tsx
Comment (aeolianeth) [inline src/components/Create/RulesForm.tsx:97]: **issue**: the "syntax" of `key: value` in some languages is different. Here's a decent explanation as to why: https://softwareengineering.stackexchange.com/questions/234546/colons-in-internationalized-ui

This can thus become:

```suggestion
                <Trans>Contract address: {s.address}</Trans>
```

File: src/components/Create/RulesForm.tsx
Comment (aeolianeth) [inline src/components/Create/RulesForm.tsx:115]: **issue** lets wrap this whole block in a `Trans` to avoid sentence splitting.

File: src/components/Dashboard/Rewards.tsx
Comment (aeolianeth) [inline src/components/Dashboard/Rewards.tsx:151]: **issue**: ideally we'd avoid splitting the sentence here for translation. Not sure if wrapping the whole thing in `t` will work though - would you mind giving it a crack?

File: src/components/Dashboard/Rewards.tsx
Comment (aeolianeth) [inline src/components/Dashboard/Rewards.tsx:231]: **issue** lets wrap this whole block in a `Trans` to avoid sentence splitting.

File: src/components/FundingCycle/FundingCycleDetails.tsx
Comment (aeolianeth) [inline src/components/FundingCycle/FundingCycleDetails.tsx:175]: **issue** lets wrap this whole block in a `Trans` to avoid sentence splitting.

File: src/components/Landing/Footer.tsx
Comment (aeolianeth) [inline src/components/Landing/Footer.tsx:34]: **praise**: awesome implementation!! love the scrollTo detail ✨

File: src/components/Navbar/Logo.tsx
Comment (aeolianeth) [inline src/components/Navbar/Logo.tsx:6]: **suggestion (non-blocking)**: we could provide a default value for height:

```suggestion
export default function Logo({ height = 40 }: { height?: number }) {
```

File: src/components/Navbar/NavLanguageSelector.tsx
Comment (aeolianeth) [inline src/components/Navbar/NavLanguageSelector.tsx:28]: **suggestion**: might be better to declare this as `const setLanguage = ()...` to avoid [hoisting](https://www.w3schools.com/js/js_hoisting.asp) weirdness.

File: src/components/Navbar/NavLanguageSelector.tsx
Comment (aeolianeth) [inline src/components/Navbar/NavLanguageSelector.tsx:8]: **praise**: well done on this implementation, and thanks for the detailed comments throughout!

File: src/components/shared/formItems/ProjectPayoutMods.tsx
Comment (aeolianeth) [inline src/components/shared/formItems/ProjectPayoutMods.tsx:365]: **issue**: lets wrap whole thing in `Trans` to avoid splitting sentence.

File: src/components/shared/formItems/ProjectTicketMods.tsx
Comment (aeolianeth) [inline src/components/shared/formItems/ProjectTicketMods.tsx:277]: **issue**: lets wrap whole thing in `Trans` to avoid splitting sentence.

File: src/styles/antd-overrides/collapse.scss
Comment (aeolianeth) [inline src/styles/antd-overrides/collapse.scss:88]: **issue (non-blocking)**: hopefully we can get to a place where there are less magic numbers regarding heights. Non blocking for this PR though :)

**Review by aeolianeth — CHANGES_REQUESTED** [review]
@johnnyd-eth sooo excited for this PR! Really great work, this is such an impactful change.

I left a few comments, mostly related to avoiding splitting sentences when marking strings for translations.

It would also be good to check this on Firefox, Safari and Edge (+ mobile for each) if you haven't already.

File: src/styles/antd-overrides/collapse.scss
Comment (johnnyd-eth) [reply src/styles/antd-overrides/collapse.scss:88]: Yep gotcha!

File: src/components/Navbar/NavLanguageSelector.tsx
Comment (johnnyd-eth) [reply src/components/Navbar/NavLanguageSelector.tsx:8]: Thanks!

File: src/components/Create/RulesForm.tsx
Comment (johnnyd-eth) [reply src/components/Create/RulesForm.tsx:97]: > **issue**: the "syntax" of `key: value` in some languages is different. Here's a decent explanation as to why: https://softwareengineering.stackexchange.com/questions/234546/colons-in-internationalized-ui
> 
> This can thus become:

Nice pickup, thanks!

File: src/components/Navbar/NavLanguageSelector.tsx
Comment (johnnyd-eth) [reply src/components/Navbar/NavLanguageSelector.tsx:28]: Good idea

File: src/components/shared/formItems/ProjectTicketMods.tsx
Comment (johnnyd-eth) [reply src/components/shared/formItems/ProjectTicketMods.tsx]: Thank you for your comments @aeolianeth! I've fixed up all those points.

**Comment by johnnyd-eth** [comment]
Thank you for your comments @aeolianeth! I've fixed up all those points.

**Comment by aeolianeth** [comment]
Dope @johnnyd-eth, approving and sending back to @peripheralist for a final look.

**Review by aeolianeth — APPROVED** [review]

**Comment by peripheralist** [comment]
I know we talked about holding off on making the mobile nav menu fullwidth, but just realized i had only looked at this with a short ens name wallet attached to the app. Looks like with a longer name or a full address it actually bleeds off the menu. maybe for this PR we can just set a fixed width of like 300 which should be wide enough for most names?

image

**Comment by johnnyd-eth** [comment]
> I know we talked about holding off on making the mobile nav menu fullwidth, but just realized i had only looked at this with a short ens name wallet attached to the app. Looks like with a longer name or a full address it actually bleeds off the menu. maybe for this PR we can just set a fixed width of like 300 which should be wide enough for most names?
> 
> image

I've just made another PR #340 which is just a better version of the mobile nav we have here. Should we just merge it into this branch and ship it? I'm sure there are still improvements to be made in the mobile nav but #340 is clearly better than we have here.

**Comment by peripheralist** [comment]
Yep @johnnyd-eth love it! #340 looks great, let's just merge that in with this and then take it to main

**Comment by johnnyd-eth** [comment]
> Yep @johnnyd-eth love it! #340 looks great, let's just merge that in with this and then take it to main

Ok donski @peripheralist

**Comment by peripheralist** [comment]
sorry @johnnyd-eth i must have forgotten to get to this before the contracts-v1.1 branch merge. let's just solve those last merge conflicts

**Comment by johnnyd-eth** [comment]
> sorry @johnnyd-eth i must have forgotten to get to this before the contracts-v1.1 branch merge. let's just solve those last merge conflicts

No worries @peripheralist. All resolved

**Comment by johnnyd-eth** [comment]
Resolved latest @peripheralist

**Pull request merged**
